### PR TITLE
Fix parameters

### PR DESCRIPTION
**Pull request by Jegp**
#57

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -54,7 +54,7 @@ Running experiments is as easy as running
 
     python3 -m norse.task.mnist
 
-This will run a simple MNIST experiment with default parameters. See :doc:`/experiments` for detailed instructions.
+This will run a simple MNIST experiment with defaultp. See :doc:`/experiments` for detailed instructions.
```
Comment (cpehle) [inline docs/source/index.rst:57]: Your replace string ran a bit wild here.

File: norse/benchmark/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*.csv
```
Comment (cpehle) [inline norse/benchmark/.gitignore:1]: I would probably just add this to the top level gitignore?

File: norse/task/cartpole.py [hunk 3]
```diff
@@ -210,7 +210,7 @@ def main():
         policy = Policy()
     elif FLAGS.policy == "lsnn":
         policy = LSNNPolicy(device=FLAGS.device, model=FLAGS.model).to(FLAGS.device)
-    optimizer = torch.optim.Adam(policy.parameters(), lr=FLAGS.learning_rate)
+    optimizer = torch.optim.Adam(policy.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/cartpole.py:213]: this should be parameters()

File: norse/task/cifar10.py [hunk 4]
```diff
@@ -153,7 +153,7 @@ def train(
         loss = torch.nn.functional.nll_loss(output, target)
         loss.backward()
         if FLAGS.save_grads and batch_idx % FLAGS.grad_save_interval == 0:
-            for idx, p in enumerate(model.parameters()):
+            for idx, p in enumerate(model.p()):
```
Comment (cpehle) [inline norse/task/cifar10.py:156]: replace string function

File: norse/task/cifar10.py [hunk 5]
```diff
@@ -358,19 +358,19 @@ def signed_current_encoder(x):
 
     if FLAGS.optimizer == "sgd":
         optimizer = torch.optim.SGD(
-            model.parameters(),
+            model.p(),
```
Comment (cpehle) [inline norse/task/cifar10.py:361]: parameters

File: norse/task/cifar10.py [hunk 6]
```diff
@@ -358,19 +358,19 @@ def signed_current_encoder(x):
...
             lr=FLAGS.learning_rate,
             momentum=0.9,
             weight_decay=5e-4 * FLAGS.batch_size,
             nesterov=True,
         )
     elif FLAGS.optimizer == "adam":
-        optimizer = torch.optim.Adam(model.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.Adam(model.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/cifar10.py:368]: parameters

File: norse/task/cifar10.py [hunk 7]
```diff
@@ -358,19 +358,19 @@ def signed_current_encoder(x):
...
     elif FLAGS.optimizer == "rms":
-        optimizer = torch.optim.RMSprop(model.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.RMSprop(model.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/cifar10.py:370]: parameters

File: norse/task/cifar10.py [hunk 8]
```diff
@@ -358,19 +358,19 @@ def signed_current_encoder(x):
...
 
     if FLAGS.only_output:
-        optimizer = torch.optim.Adam(model.out.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.Adam(model.out.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/cifar10.py:373]: parameters

File: norse/task/correlation_experiment.py [hunk 9]
```diff
@@ -65,9 +65,7 @@ def generate_random_data(
     rec_linear_update = torch.nn.Linear(2 * 8 * 8, 8 * 8)
 
     optimizer = torch.optim.Adam(
-        list(linear_update.parameters())
-        + [input_weights, recurrent_weights]
-        + list(out.parameters()),
+        list(linear_update.p()) + [input_weights, recurrent_weights] + list(out.p()),
```
Comment (cpehle) [inline norse/task/correlation_experiment.py:68]: parameters

File: norse/task/mnist.py [hunk 10]
```diff
@@ -115,7 +115,7 @@ def train(model, device, train_loader, optimizer, epoch, writer=None):
         loss.backward()
 
         if FLAGS.clip_grad:
-            torch.nn.utils.clip_grad_norm_(model.parameters(), FLAGS.grad_clip_value)
+            torch.nn.utils.clip_grad_norm_(model.p(), FLAGS.grad_clip_value)
```
Comment (cpehle) [inline norse/task/mnist.py:118]: parameters

File: norse/task/mnist.py [hunk 11]
```diff
@@ -282,12 +282,12 @@ def main():
     ).to(device)
 
     if FLAGS.optimizer == "sgd":
-        optimizer = torch.optim.SGD(model.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.SGD(model.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/mnist.py:285]: parameters

File: norse/task/mnist.py [hunk 12]
```diff
@@ -282,12 +282,12 @@ def main():
...
     elif FLAGS.optimizer == "adam":
-        optimizer = torch.optim.Adam(model.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.Adam(model.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/mnist.py:287]: parameters

File: norse/task/mnist.py [hunk 13]
```diff
@@ -282,12 +282,12 @@ def main():
...
 
     if FLAGS.only_output:
-        optimizer = torch.optim.Adam(model.out.parameters(), lr=FLAGS.learning_rate)
+        optimizer = torch.optim.Adam(model.out.p(), lr=FLAGS.learning_rate)
```
Comment (cpehle) [inline norse/task/mnist.py:290]: parameters

File: norse/torch/functional/encode.py [hunk 14]
```diff
@@ -35,7 +35,7 @@ def constant_current_lif_encode(
     Parameters:
         input_current (torch.Tensor): The input tensor, representing LIF current
         seq_length (int): The number of iterations to simulate
-        parameters (LIFParameters): Initial neuron parameters. Defaults to zero.
+        p (LIFParameters): Initial neuronp. Defaults to zero.
```
Comment (cpehle) [inline norse/torch/functional/encode.py:38]: neuron parameters

File: norse/torch/module/encode.py [hunk 15]
```diff
@@ -31,20 +31,17 @@ def __init__(
 
         Parameters:
             seq_length (int): The number of iterations to simulate
-            parameters (LIFParameters): Initial neuron parameters. Defaults to zero.
+            p (LIFParameters): Initial neuronp. Defaults to zero.
```
Comment (cpehle) [inline norse/torch/module/encode.py:34]: neuron parameters

File: docs/source/index.rst (hunk 1 above)
Comment (Jegp) [reply docs/source/index.rst:57]: Argh, great catch.

**Comment by Jegp** [comment]
Yeah, a lot of damage. Thanks for catching this! I need to review this tomorrow when I'm less tired. Ideally we should have simple tests for all of these...

**Pull request merged**